src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,30 +33,32 @@ export const TodoModal: React.FC = () => {
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
```

Comment:
Replace `{`Todo #${selectedTodo}`}` with `{`Todo #${selectedTodo.id}`}` to display the correct Todo ID in the modal header.